jenkins/scripts/perf/local/submit.py

```
@@ -0,0 +1,530 @@
+            args.test_list, llm_src, is_disagg
+        )
+    elif args.config_file:
+        config_yaml = args.config_file
+        config_file_base_name = os.path.splitext(os.path.basename(config_yaml))[0]
+        test_name = args.test_name if not is_disagg else None
+    else:
+        raise ValueError("Either --test-list or --config-file must be provided")
+
+    # Validate test_name for aggregated mode
+    if not is_disagg and not test_name:
+        raise ValueError("--test-name is required for aggregated mode when --config-file is used")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)